### PR TITLE
Use scmVars before env variables for git url detection

### PR DESCRIPTION
**Pull request by soliad**
Needed for organisation or multibranch pipelines

### CODE REVIEW

**Comment by soliad** [comment]
Command to use in the organisation/multibranch pipeline when GIT_URL is not propagated properly:
`step([$class: 'MasterCoverageAction', scmVars: [GIT_URL: env.GIT_URL]])`
`step([$class: 'CompareCoverageAction', scmVars: [GIT_URL: env.GIT_URL]])`

File: src/main/java/com/github/terma/jenkins/githubprcoveragestatus/PrIdAndUrlUtils.java [hunk 1]
```diff
@@ -79,9 +79,9 @@ public static String getGitUrl(final Map<String, String> scmVars, final Run buil
         Map<String, String> envVars = build.getEnvironment(listener);
         final String gitUrl = envVars.get(GIT_URL_PROPERTY);
         final String changeUrl = envVars.get(CHANGE_URL_PROPERTY);
-        if (gitUrl != null) return gitUrl;
+        if (scmVars != null && scmVars.containsKey(GIT_URL_PROPERTY)) return scmVars.get(GIT_URL_PROPERTY);
+        else if (gitUrl != null) return gitUrl;
```
Comment (terma) [inline src/main/java/com/github/terma/jenkins/githubprcoveragestatus/PrIdAndUrlUtils.java]: Please add test

Comment (soliad) [reply src/main/java/com/github/terma/jenkins/githubprcoveragestatus/PrIdAndUrlUtils.java]: Done

**Comment by Spea** [comment]
I there anything I can do to help getting this merged? Would really love to have this feature in order to setup our code-coverage reporting.

**Comment by estelop** [comment]
Ok, new year :)
Any chance this will be merged and released soon? This issue is the only thing preventing me from using the plugin

**Comment by gokhurana** [comment]
Is there anything blocking it from getting merged?

**Comment by terma** [comment]
I'll take a look today

**Comment by gokhurana** [comment]
Thanks! @terma . It will be really helpful if we could get it in. unless there's an issue. Looking at the PR it seems pretty clean though.
Can you please get it merged soon? Thanks much!!

**Comment by gokhurana** [comment]
@terma this will actually unblock many of us and without this I'll have to write an ugly code in my Jenkinsfile to pull the master coverage and show the comparison.

**Comment by gokhurana** [comment]
Thanks much!  Is the plugin going to be available for Jenkins 2.89.3?

**Comment by terma** [comment]
Released under version [1.10.0](https://github.com/jenkinsci/github-pr-coverage-status-plugin/releases/tag/github-pr-coverage-status-1.10.0)

**Comment by terma** [comment]
@gokhurana yep should be available for all version of ```Jenkins > 2```

**Comment by gokhurana** [comment]
Thanks! Installed and works great!

**Pull request merged**